File: src/packets/PacketS2CProperty.js
const PacketS2C = require("./PacketS2C");
const Property = require("./Property");

/**
 * Server -> Client
 * ID = 0x02
 * Property
 * Sends the client a controller property value
 */
class PacketS2CProperty extends PacketS2C {
    constructor(property, value) {
        super();

        this._buffer.writeUInt8(property);
        switch (property)
        {
        case Property.PROPERTY_SERIAL_CONNECTED:
        case Property.PROPERTY_WEB_SERVER_ENABLED:
            this._buffer.writeUInt8(value === true ? 1 : 0);
            break;
        case Property.PROPERTY_NAME:
        case Property.PROPERTY_VERSION:
            this._buffer.writeStringNT(value);
            break;
        }
    }

    getID() {
        return 0x02;
    }
}

module.exports = PacketS2CProperty;

File: src/RNetPi.Core/Packets/PacketS2CProperty.cs
using RNetPi.Core.Constants;

namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x02
/// Property
/// Sends the client a controller property value
/// Data:
///     (Unsigned Char) Property ID
///     (Variable) Property Value
/// </summary>
public class PacketS2CProperty : PacketS2C
{
    public const byte ID = 0x02;

    public PacketS2CProperty(byte property, object? value)
    {
        Writer.Write(property);
        switch (property)
        {
            case Properties.SerialConnected:
            case Properties.WebServerEnabled:
                Writer.Write((byte)((value is bool boolValue && boolValue) ? 1 : 0));
                break;
            case Properties.Name:
            case Properties.Version:
                WriteNullTerminatedString(value?.ToString() ?? string.Empty);
                break;
        }
    }

    public override byte GetID() => ID;
}